app/src/main/java/paulscode/android/mupen64plusae/GalleryActivity.java

```diff
@@ -575,6 +646,22 @@ public void onPause() {
 
         super.onPause();
 
+        // We don't resume resetting the game if something happens here because we
+        // set the gallery grid to a blank state which could be saved if not loaded
+        // properly. So to avoid the issue we just stop the reset
+        if(mGameResetting){
+            // Could retain and show notification onResume but the data might not be retained
+            // so we'd have to write to disk which becomes a problem because the user could
+            // exit for good, we wouldn't know which could cause some confusing/false notifications
+            if(isChangingConfigurations())
+                runOnUiThread(() -> Notifier.showToast(this, R.string.toast_unexpectedChange));
```

Comment:
onPause should already be running on the UI thread, so you shouldn't have to call runOnUiThread here.